service/src/test/java/uk/nhs/adaptors/gpc/consumer/gpc/SdsFilterTest.java

```diff
@@ -0,0 +1,101 @@
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.cloud.gateway.filter.GatewayFilterChain;
+import org.springframework.mock.http.server.reactive.MockServerHttpRequest;
+import org.springframework.mock.web.server.MockServerWebExchange;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+import uk.nhs.adaptors.gpc.consumer.filters.SdsFilter;
+import uk.nhs.adaptors.gpc.consumer.sds.SdsClient;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class SdsFilterTest {
+
+    @Mock
+    private SdsClient sdsClient;
+    private SdsFilter sdsFilter;
+    private GatewayFilterChain filterChain;
+    private ArgumentCaptor<ServerWebExchange> captor;
+
+    private static final String MIGRATE_STRUCTURED_INTERACTION
+                                                    = "urn:nhs:names:services:gpconnect:fhir:operation:gpc.migratestructuredrecord-1";
+
+    @BeforeEach
+    @SneakyThrows
+    public void before() {
+        sdsFilter = new SdsFilter(sdsClient);
+        sdsFilter.initializeSdsRequestFunctions();
+
+        filterChain = Mockito.mock(GatewayFilterChain.class);
+        captor = ArgumentCaptor.forClass(ServerWebExchange.class);
+        when(filterChain.filter(captor.capture())).thenReturn(Mono.empty());
+    }
+
+    @Test
+    @SneakyThrows
+    public void When_NoSspHeaders_Expect_FetchValuesFromSds() {
+
+        String odsCode = "A12345";
+        String correlationId = "98765";
+        String gpConnectServerAsid = "928940000057";
+        String gpConnectConsumerAsid = "928942012545";
+
+        MockServerHttpRequest request
+                        = MockServerHttpRequest.get("/A12345/STU3/1/gpconnect/fhir/Patient/$gpc.migratestructuredrecord")
+                        .header("Ssp-TraceID", correlationId)
+                        .header("Ssp-InteractionID",
+                                "urn:nhs:names:services:gpconnect:fhir:operation:gpc.migratestructuredrecord-1").build();
+        MockServerWebExchange exchange = MockServerWebExchange.from(request);
+
+        when(sdsClient.callForGetAsid(MIGRATE_STRUCTURED_INTERACTION, odsCode, correlationId)).thenReturn(gpConnectConsumerAsid);
+        when(sdsClient.callForMigrateStructuredRecord(odsCode, correlationId))
+                                    .thenReturn(Mono.just(SdsClient.SdsResponseData.builder().nhsSpineAsid(gpConnectServerAsid).build()));
+
+        sdsFilter.filter(exchange, filterChain);
+
+        var resultExchange = captor.getValue();
+        assertEquals(gpConnectConsumerAsid, resultExchange.getRequest().getHeaders().get("ssp-From").get(0));
+        assertEquals(gpConnectServerAsid, resultExchange.getRequest().getHeaders().get("ssp-To").get(0));
+    }
+
+    @Test
+    @SneakyThrows
+    public void When_SspValuesHeadersPresent_Expect_FetchSspValuesFromRequestHeaders() {
```

Comment:
Given that the current behaviour of the GP2GP adaptor is to provide an `Ssp-From` header, and this repo only adds the header if it isn't already provided, we'll need to create a PR in the GP2GP to no longer provide the header.

This will cause an interesting breaking change to the GP2GP adaptor where we'll need to be explicit that a specific version of the GPConnect Consumer adaptor is required.

Reply:
the changes are done to remove "ssp-from" header from GP2GP app.